pkg/batch/executor_test.go

```diff
@@ -11,6 +11,50 @@ import (
 	"github.com/treeverse/lakefs/pkg/logging"
 )
 
+const accessedOnce = 1
+const accessedTwice = 2
+
+type reader struct {
+	Ch         chan bool
+	IsExecuted bool
+	Fn         ExecFn
+}
+
+type ExecFn func() (interface{}, error)
+
+func (r *reader) Execute() (interface{}, error) {
+	r.IsExecuted = true
+	return r.Fn()
+}
+
+func (r *reader) Batched() {
+	close(r.Ch)
+}
+
+type db struct {
+	KvStore     sync.Map
+	AccessCount int
+}
+
+type Dao interface {
+	Insert(key string, val string)
+	Get(key string) string
+	GetCount() int
+}
+
+func (d *db) Insert(key string, val string) {
+	d.KvStore.Store(key, val)
+}
+
+func (d *db) Get(key string) (interface{}, bool) {
+	d.AccessCount++
```

Comment:
Given that `KvStore` is a `sync.Map`, this is a race.  You might need to perform an atomic action here, or just lock the whole `db` and do away with the `sync.Map`.